lib/wisper/active_job_broadcaster.rb

```diff
@@ -5,7 +5,11 @@
 module Wisper
   class ActiveJobBroadcaster
     def broadcast(subscriber, publisher, event, args)
-      Wrapper.perform_later(subscriber.name, event, args)
+      if subscriber < ActiveJob::Listener
+        subscriber.perform_later(event, args)
+      else
+        Wrapper.perform_later(subscriber.name, event, args)
+      end
```

Comment:
If we have enforced everyone to use there own class do we need this? and do we need the wrapper?

Reply:
We have not enforced that. In the current implementation, `Wisper::ActiveJob::Listener` is optional, so we still need the Wrapper for classes that don't include `Wisper::ActiveJob::Listener`.

For one, if this is going to get upstreamed, we have to maintain backwards compatibility, and even in our own internal PR we aren't using this everywhere from the get-go, so we also need that backwards compat.

Make sense?

Reply:
:+1: